dashboard/src/shared/Auth.ts

```diff
@@ -68,21 +76,35 @@ export class Auth {
     }
   }
 
-  // isAuthenticatedWithCookie() does a HEAD request (without any token obviously)
-  // to determine if the request is authenticated (ie. not a 401). Unfortunately
-  // Kubernetes defaulting to allow anonymous requests means that this will be a 403
-  // even if there are no credentials, so we additionally check the message and assume
-  // we are authenticated with a cookie if a 403 is not for an anon user.
+  // isAuthenticatedWithCookie() does an anonymous GET request to determine if
+  // the request is authenticated with an http-only cookie (there is, by design,
+  // no way to determine via client JS whether an http-only cookie is present).
   public static async isAuthenticatedWithCookie(): Promise<boolean> {
     try {
       await Axios.get(APIBase + "/");
     } catch (e) {
       const response = e.response as AxiosResponse;
+      // The only error response which can possibly mean we did authenticate is
+      // a 403 from the k8s api server (ie. we got through to k8s api server
+      // but RBAC doesn't authorize us).
+      if (response.status !== 403) {
+        return false;
+      }
+      // A non json 403 error response means we did not get through to the API server but
+      // instead were rejected by the auth proxy (ie. no http-only cookie).
+      // TODO(mnelson): Check why doesn't the auth proxy return a 401 for a request without auth?
+      if (response.headers && response.headers["content-type"] !== "application/json") {
+        return false;
+      }
```

Comment:
is this condition needed? If the response is not an `application/json`, below `response.data.message` will be undefined, right?

Reply:
> If the response is not an application/json, below response.data.message will be undefined, right?

Yes, which would mean that isAnon would be false and the current code assumes it is authed (ie. a 403 that isn't anonymous). I could replace the above condition with one based on `response.data` as you say  - perhaps that'd be simpler? Something like:
```
if (response.data === undefined) {
  return false;
}
const isAnon = response.data.message && ...
```

Reply:
ah, okay, for a moment I thought that with `isAnon = false` we were returning `false`.

Then yes, I'd rather check if the response has data  (and a messsage):

```
if (!response.data || !response.data.message) {
  return false;
}
``` 

Also this allows you to simple return `return response.data.message.includes("system:anonymous");` below since we are already checking that `data` and `message` are defined.

Reply:
Yep, updated - that's neater.



> Also this allows you to simple return `return response.data.message.includes("system:anonymous");` below since we are already checking that `data` and `message` are defined.

fwiw, I've still left the const, so that the return reads as "return not anonymous":
```
const isAnon = response.data.message.includes("system:anonymous");
return !isAnon;
```

Thanks